### ISSUE TITLE
%%javascript does not execute properly

### ISSUE THREAD

**Issue by parente** [jupyter-incubator/dashboards_nodejs_app#82]
Try https://github.com/jupyter-incubator/dashboards/blob/master/etc/notebooks/test/test_html_javascript.ipynb in the dashboard server. Works fine in the notebook server and the old Thebe setup. Here, the %%javascript cell does not render as a script tag but rather as a javascript object string.

Might be a bug in jupyter-js-output-area or transformime. Might be a bug in how we consume messages.

**Comment by nitind** [comment]
I'll dig into this.

**Comment by parente** [comment]
test_html_javascript.ipynb has the %%javascript cell that I tested when I opened the issue. 

test_layout.ipynb also emits javascript, but not using the magic. It uses the matplotlib interactive plots which emit JS on a regular python call. I see `<IPython.core.display.Javascript object>` in the page which is more evidence that the JS is not being transformed properly into a script tag.

PS - I doubt the matplotlib JS will work properly even when it does land in the page as a proper script. So don't worry about debugging that yet.

**Comment by parente** [comment]
The problem is that the current release of `transformime-jupyter-transformers` does not consider mimetype `application/javascript` for its ScriptTransformer. The latest master branch in the project has the fix since Nov 11, but is not yet in an npm release. Once that release is made, `jupyter-js-output-area` will pick it up because of the version spec in its package.json here https://github.com/jupyter/jupyter-js-output-area/blob/master/package.json#L61.

@rgbkrk Can you cut the 0.2.1 release of transformime-jupyter-transformers?

**Comment by rgbkrk** [comment]
Whoop, I cut it as 0.3.0 since it added new support.

Also made a PR to jupyter js output area:

https://github.com/jupyter/jupyter-js-output-area/pull/32

**Comment by blink1073** [comment]
@parente, FYI, we are deprecating `jupyter-js-input-area`, `jupyter-js-output-area`, and `jupyter-js-cells`, to be instead called as `require('jupyter-js-notebook/lib/input-area')`, etc.

**Comment by parente** [comment]
@blink1073 Yea, we were going to move to jupyter-js-notebook in #67 to baby step this. But looks like now's the time to jump to get this fix.

**Comment by rgbkrk** [comment]
Well now you have it in two places so you can mix and match until you transition!

**Comment by parente** [comment]
@rgbkrk :+1:  and :+1: (?)

**Comment by rgbkrk** [comment]
I'm all thumbs. Just want to make sure you're not blocked on anything and can keep iterating.

**Comment by parente** [comment]
@nitind is testing to see if we're all set. Thanks.

**Comment by parente** [comment]
Fixed by #105 which also bumped us to using jupyter-js-notebook.

Sidenote: while `%%javascript` cells now worker properly here, things like `%matplotlib notebook` rendering does not. While we now run the script that matplotlib emits, its JS makes lots of references to `IPython.notebook` like so:

```javascript
if (IPython.notebook.kernel != null) {
    IPython.notebook.kernel.comm_manager.register_target('matplotlib', mpl.mpl_figure_comm);
}
```

I don't know if we want to keep adding shims for notebook references. We already have for declarative widgets. We might want to look at some of the common plotting libs (plot.ly, bokeh, matpotlib) in more depth to see if providing a small set would take us a long ways as these libraries transition.
